opal-shell/src/main/java/org/obiba/opal/shell/commands/AnalyseCommand.java

```diff
@@ -115,6 +125,59 @@ public int execute() {
     return 0;
   }
 
+  private class AnalysisValueTableResolver {
+    ValueTable view = null;
+
+    public ValueTable resolve(ValueTable source, String variableNamesCsv) {
+      List<String> variableNames = getVariableNames(variableNamesCsv);
+      if (variableNames.isEmpty()) return source;
+
+      view = createView(source, variableNames);
+      log.debug("Created view {} for variables {}", view.getName(), variableNamesCsv);
+
+      return view;
+    }
+
+    private ValueTable createView(ValueTable table, List<String> variableNames) {
+      String viewName = table.getName() + "View";
+      View.Builder builder = View.Builder.newView(viewName, Lists.newArrayList(table));
+      List<Variable> variables = variableNames.stream()
+        .filter(variableName -> table.hasVariable(variableName))
+        .map(variableName -> {
+          Variable variable = table.getVariable(variableName);
+          return Variable.Builder
+            .sameAs(variable, true)
+            .addAttribute("script", String.format("$('%s')", variableName))
+            .build();
+        })
+        .collect(Collectors.toList());
```

Comment:
Executing script for each variable value is expensive (there is an overhead when setting up the javascript execution context). Use the SelectClause instead, because you just want to filter the variables.